src/mlpack/methods/linear_svm/linear_svm.hpp

```diff
@@ -0,0 +1,211 @@
+/**
+ * @file linear_svm.hpp
+ * @author Ayush Chamoli
+ *
+ * An implementation of Linear SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_LINEAR_SVM_LINEAR_SVM_HPP
+#define MLPACK_METHODS_LINEAR_SVM_LINEAR_SVM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <ensmallen.hpp>
+
+#include "linear_svm_function.hpp"
+
+namespace mlpack {
+namespace svm {
+
+/**
+ * The LinearSVM class implements an L2-regularized support vector machine
+ * model, and supports training with multiple optimizers and classification.
+ * The class supports different observation types via the MatType template
+ * parameter; for instance, support vector classification can be performed
+ * on sparse datasets by specifying arma::sp_mat as the MatType parameter.
+ *
+ * Linear SVM can be used for general classification tasks which will work
+ * on multiclass classification. More technical details about
+ * the model can be found on the following webpage:
+ *
+ * https://link.springer.com/content/pdf/10.1007%2FBF00994018.pdf
```

Comment:
It doesn't seem like this details the multiclass linear SVM extension you gave me in https://www.elen.ucl.ac.be/Proceedings/esann/esannpdf/es1999-461.pdf .  Should we reference both of those here?  (Also we should use BibTeX for the references, since direct URL links may not always work.)

Reply:
I'll make the changes here, thanks for pointing this out.

Reply:
I'm not sure how I should reference these both using BibTeX, I looked at `gan.hpp` and `dtree.hpp` (for a reference) but it seems like both of those represented in different manner since the later one was published in a book. Could you help me out in this?